File: src/main/java/com/api/prisma_vi/user/auth/register/RegisterForm.java
package com.api.prisma_vi.user.auth.register;

import com.api.prisma_vi.user.UserRole;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record RegisterForm(

        @NotBlank(message = "Nickname cannot be blank")
        String nickname,

        @Email(message = "Invalid email") @NotBlank(message = "Invalid cannot be blank")
        String email,

        @NotBlank(message = "Password cannot be blank") @Size(min = 6, max = 16, message = "Password must contain 6 to 16 characters")
        String password,

        UserRole role
) {
}

File: src/main/java/com/api/prisma_vi/auth/register/RegisterForm.kt
package com.api.prisma_vi.auth.register

import com.api.prisma_vi.user.UserRole
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank

data class RegisterForm(

    @field:NotBlank(message = "Nickname cannot be blank")
    val nickName: String,

    @field:Email(message = "Invalid email") @field:NotBlank(message = "Invalid cannot be blank")
    val email: String,

    @field:NotBlank(message = "Password cannot be blank")
    val password: String,

    val role: UserRole? = null

)
